go/parquet/compress/compress_test.go

```diff
@@ -0,0 +1,128 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compress_test
+
+import (
+	"bytes"
+	"io/ioutil"
+	"math/rand"
+	"testing"
+
+	"github.com/apache/arrow/go/parquet/compress"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	RandomDataSize       = 3 * 1024 * 1024
+	CompressibleDataSize = 8 * 1024 * 1024
+)
+
+func makeRandomData(size int) []byte {
+	ret := make([]byte, size)
+	r := rand.New(rand.NewSource(1234))
+	r.Read(ret)
+	return ret
+}
+
+func makeCompressibleData(size int) []byte {
+	const base = "Apache Arrow is a cross-language development platform for in-memory data"
+
+	data := make([]byte, size)
+	n := copy(data, base)
+	for i := n; i < len(data); i *= 2 {
+		copy(data[i:], data[:i])
+	}
+	return data
+}
+
+func TestCompressDataOneShot(t *testing.T) {
+	tests := []struct {
+		c compress.Compression
+	}{
+		{compress.Codecs.Uncompressed},
+		{compress.Codecs.Snappy},
+		{compress.Codecs.Gzip},
+		// {compress.Codecs.Lzo},
+		{compress.Codecs.Brotli},
+		{compress.Codecs.Lz4},
```

Comment:
LZ4 is problematic: http://mail-archives.apache.org/mod_mbox/arrow-dev/202007.mbox/%3CCAAri41v24xuA8MGHLDvgSnE+7AAgOhiEukemW_oPNHMvfMmrWw@mail.gmail.com%3E

Reply:
I saw the mentions between LZ4 and the Hadoop LZ4 in the codebase. What is actually the difference in the algorithm / should I remove/disable the LZ4 handling here until i can specifically get the "Hadoop LZ4" hooked up?

Reply:
It is a little bit of a rabbit whole.  I would consider just saying this isn't supported.  

LZ4 has two specified formats: "raw" (no additional metadata) and "framed" (provides metadata/checksums/etc).  For a while C++ made use of framed I think.  This was incompatible.  An effort was made to try to use "raw" in what we thought was the Hadoop LZ4 format (turns out it wasn't).  There is a new proposal to use "raw" which will subsume LZ4 Hadoop.

I think hadoop is its own unspecified (except in code) framing on top of raw.

Reply:
Ah interesting. Ok, I'll leave it as unsupported for now and if "raw" subsumes LZ4 Hadoop in the parquet spec I'll implement that at that time. I'll update the PR.